vmslave-config/pipeline/vmslave_config.groovy

```diff
@@ -0,0 +1,29 @@
+def checkout(String url, String branch, String targetDir){
+    checkout(
+    [$class: 'GitSCM', branches: [[name: branch]],
+    extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: targetDir]],
+    userRemoteConfigs: [[url: url]]])
+}
+
+node('vmslave-config') {
+    withEnv([
+        "GIT_PREVIOUS_COMMIT=${env.GIT_PREVIOUS_COMMIT}",
```

Comment:
In jenkins job we need a true var list to run ansible-play book.
But we can't show this value in codes.
So
these k=v are defined in jenkins job parameters.
```withEnv``` provides these vars to ```vmslave_config.sh``` with use the ```pipeline/all``` var file.
You can see the ```pipeline/all``` file for more information